From: Mathias Kresin <dev@kresin.me>
Date: Fri, 14 Jul 2017 19:35:02 +0200
Subject: image: fix ar71xx legacy images

If TARGET_PER_DEVICE_ROOTFS and DEVICE_PACKAGES are used for ar71xx
legacy images:

- an already jffs2 padded squashfs rootfs is overwritten
  with an unpadded/raw one.

- the squashfs-raw and squashfs-64k rootfs are not replaced by the
  ones including the DEVICE_PACKAGES

Call Image/Build/squashfs after the DEVICE_PACKAGES are added to the
base squashfs rootfs to fix the issues.

Fixes: FS#904

Signed-off-by: Mathias Kresin <dev@kresin.me>

Origin: upstream, https://git.lede-project.org/?p=source.git;a=commitdiff;h=f6907dcc7968a81eb
Bug-gluon: https://github.com/freifunk-gluon/gluon/issues/1193

diff --git a/include/image-legacy.mk b/include/image-legacy.mk
index 1864d714e55ccefc0e6c3f9c12fde14d0f6080f6..1ccaec09b04674294db48ba2ec1ae3679544b4f1 100644
--- a/include/image-legacy.mk
+++ b/include/image-legacy.mk
@@ -48,6 +48,7 @@ endef
 ifdef TARGET_PER_DEVICE_ROOTFS
   define Image/Build/Profile/Filesystem
 	cp $(KDIR)/root.$(2)+pkg=$(3) $(KDIR)/root.$(2)
+	$(call Image/Build/$(2),$(2))
 	$(call Image/Build/Profile,$(1),$(2))
   endef
 else
